packages/integrations/preact/src/client.ts

```diff
@@ -0,0 +1,29 @@
+import type { SignalLike } from './types';
+import { h, render } from 'preact';
+import StaticHtml from './static-html.js';
+
+const sharedSignalMap: Map<string, SignalLike> = new Map();
+
+export default (element: HTMLElement) =>
+	async (Component: any, props: Record<string, any>, { default: children, ...slotted }: Record<string, any>) => {
+		if (!element.hasAttribute('ssr')) return;
+		for (const [key, value] of Object.entries(slotted)) {
+			props[key] = h(StaticHtml, { value, name: key });
+		}
+		let signalsRaw = element.dataset.preactSignals;
+		if(signalsRaw) {
+			const { signal } = await import('@preact/signals');
```

Comment:
Dynamic import so that non-signal users don't have this added to their bundle size.